### PR TITLE
Support for Alternate Names if using @SerializedName annotation + Consistent Order while serializing/deserializing

### PR DESCRIPTION
**Pull request by anirudhramanan**
Changes 

1. Support for Alternate Names if using @SerializedName annotation : 

screen shot 2017-01-12 at 12 28 37 pm

2. Consistent Order while Serializing / Deserializing

### CODE REVIEW

**Comment by anirudhramanan** [comment]
@anthonycr : We have updated the readme doc, listed out all the major features/changes that went in. Let us know if we missed out something here. thanks!

File: stag-library-compiler/src/main/java/com/vimeo/stag/processor/generators/model/AnnotatedClass.java [hunk 1]
```diff
@@ -24,6 +24,7 @@
 package com.vimeo.stag.processor.generators.model;
 
 import com.google.gson.annotations.SerializedName;
+import com.sun.org.apache.bcel.internal.generic.Type;
```
Comment (anthonycr) [inline stag-library-compiler/src/main/java/com/vimeo/stag/processor/generators/model/AnnotatedClass.java]: This looks like a stray import

File: stag-library-compiler/src/main/java/com/vimeo/stag/processor/utils/TypeUtils.java [hunk 2]
```diff
@@ -286,11 +286,11 @@ public static TypeMirror getInheritedType(@Nullable Element element) {
      * @param members           the member variable map of the field (Element) to their concrete
      *                          type (TypeMirror). This should be retrieved by calling getConcreteMembers
      *                          on the inherited class.
-     * @return returns a map of the member variables mapped to their concrete types for the concrete
-     * inherited class.
+     * @return returns a linked has map of the member variables mapped to their concrete types for the concrete
```
Comment (anthonycr) [inline stag-library-compiler/src/main/java/com/vimeo/stag/processor/utils/TypeUtils.java]: misspelling "hashmap"

File: stag-library-compiler/src/main/java/com/vimeo/stag/processor/generators/model/AnnotatedClass.java [hunk 3]
```diff
@@ -83,20 +83,29 @@
...
             }
         }
 
+        for (Element enclosedElement : element.getEnclosedElements()) {
+            addToSupportedTypes(enclosedElement, fieldOptions, variableNames);
+        }
+
+    }
+
+    private void addMemberVariable(@NotNull Element element, @NotNull TypeMirror typeMirror, @NotNull Map<String, Element> variableNames) {
```
Comment (anthonycr) [inline stag-library-compiler/src/main/java/com/vimeo/stag/processor/generators/model/AnnotatedClass.java]: good that we are breaking this out into it's method 👍

File: stag-library-compiler/src/main/java/com/vimeo/stag/processor/generators/TypeAdapterGenerator.java [hunk 4]
```diff
@@ -225,8 +225,16 @@ private static MethodSpec getReadMethodSpec(@NotNull TypeName typeName, @NotNull
...
+            builder.addCode("\t\t\tcase \"" + name + "\":\n");
+
+            String[] alternateJsonNames = getAlternateJsonNames(element.getKey());
+            if (alternateJsonNames != null && alternateJsonNames.length > 0) {
+                for (String alternateJsonName : alternateJsonNames) {
+                    builder.addCode("\t\t\tcase \"" + alternateJsonName + "\":\n");
+                }
+            }
+
+            builder.addCode("\t\t\t\tobject." + variableName + " = " + adapterFieldInfo.getAdapterAccessor(elementValue) + ".read(reader);");
```
Comment (anthonycr) [inline stag-library-compiler/src/main/java/com/vimeo/stag/processor/generators/TypeAdapterGenerator.java]: Awesome!

File: sample-model/src/main/java/com/vimeo/sample_model/AlternateNameModel.java [hunk 5]
```diff
@@ -0,0 +1,14 @@
+package com.vimeo.sample_model;
+
+import com.google.gson.annotations.SerializedName;
+import com.vimeo.stag.UseStag;
+
+@UseStag
+public class AlternateNameModel {
```
Comment (anthonycr) [inline sample-model/src/main/java/com/vimeo/sample_model/AlternateNameModel.java]: 👍

File: sample-model/src/main/java/com/vimeo/sample_model/AlternateNameModel1.java [hunk 6]
```diff
@@ -0,0 +1,13 @@
+package com.vimeo.sample_model;
+
+import com.google.gson.annotations.SerializedName;
+import com.vimeo.stag.UseStag;
+
+@UseStag
+public enum AlternateNameModel1 {
```
Comment (anthonycr) [inline sample-model/src/main/java/com/vimeo/sample_model/AlternateNameModel1.java]: 👍

**Comment by anirudhramanan** [comment]
@anthonycr : thanks for the review! will push the requested changes in 5-10 mins

**Comment by anirudhramanan** [comment]
done. Checked in

**Comment by anthonycr** [comment]
merged 💯

**Pull request merged**